src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,58 @@
+import { Interface } from '@ethersproject/abi'
+import { Token } from '@uniswap/sdk-core'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const tokenList = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () =>
+      Array.from(
+        new Set(positions.reduce((acc, position) => acc.concat(position.token0, position.token1), [] as string[]))
+      ).filter((address) => !tokenList[address]),
+    [positions, tokenList]
+  )
+
+  const results = useMultipleContractSingleData(nonListPositionTokenAddresses, ERC20Interface, 'symbol')
+  const addressesToSymbol: Record<string, string | undefined> = useMemo(
+    () =>
+      nonListPositionTokenAddresses.reduce(
+        (acc, address, i) => ({
+          ...acc,
+          [address]: results[i].result as string | undefined,
```

Comment:
When can `result` be undefined? Symbol not found? Call failed? Maybe we shouldn't include it in the object at all then? That way, check for presence instead of `undefined`.